lib/matplotlib/collections.py

```diff
@@ -545,9 +545,9 @@ def set_offsets(self, offsets):
         offsets = np.asanyarray(offsets)
         if offsets.shape == (2,):  # Broadcast (2,) -> (1, 2) but nothing else.
             offsets = offsets[None, :]
-        self._offsets = np.column_stack(
-            (np.asarray(self.convert_xunits(offsets[:, 0]), float),
-             np.asarray(self.convert_yunits(offsets[:, 1]), float)))
+        self._offsets = np.ma.column_stack(
```

Comment:
```suggestion
        self._offsets = np.column_stack(
```

Reply:
I understand your concern but using `np.column_stack` loses all the mask information if `offsets` is masked (the reason for this PR). I guess I could put it in an if condition to separate the two cases to make sure that masked arrays are used only when the inputs are masked and not otherwise. How does that sound?

Reply:
There is (at least) one other case where the conditional approach is used for exactly this, so I think it makes sense here as well.

Reply:
Ahh, I didn't realize that `np.column_stack` didn't handle the mask properly. It is a bit of a mess in numpy that needs sorting out over there unfortunately.
See a PR where I tried to make it a bit more uniform in the handling of all these functions: https://github.com/numpy/numpy/pull/16022 but it got reverted and I never had the time to finish it up. Maybe this will give me the motivation to revisit that.

Yes, I do think that special casing masked vs normal array is the right way to go here. Another check you can put in as a test is to verify the types of `set_offsets() == get_offsets()` so we aren't changing an input numpy array into a MaskedArray as well.

Reply:
@greglucas what is the community feeling on masked arrays?  My impression is that practically they are not used, in favour of NaN for missing/bad data.  Xarray, which is the basis for much of my analysis, doesn't even have the concept of masked arrays. 

More practically I think some of this should wait for whatever unit refactor will happen. How we input this and represent this internally should be handled by the same machinery and I'm pretty negative about casting everything to masked arrays. If anything I'd go the other way and strip the mask and enter NaN.

Reply:
> what is the community feeling on masked arrays?

My impression is that everyone thinks there should be something better/more robust than `np.ma.MaskedArray`, but it may be tricky to make a transition over to a replacement version because of some of the baked in assumptions that are already in the module.

> How we input this and represent this internally should be handled by the same machinery and I'm pretty negative about casting everything to masked arrays.

I completely agree. I don't like casting things to MaskedArrays by default either (hence my block on this as it currently stands).

> If anything I'd go the other way and strip the mask and enter NaN.

I also don't like this approach because we are changing the user's data type again. I'm hoping this can be solved by the Data Refactor work as well if we just point to a user's data rather than copy it somewhere new. A reason not to do default `NaN` casting is because then you run into another wall where we probably use `np.mean(arr)` throughout the library rather than `np.nanmean(arr)` (and similar non-nan functions) and I don't think we want to change all of those call sites around either.

Reply:
> More practically I think some of this should wait for whatever unit refactor will happen. How we input this and represent this internally should be handled by the same machinery and I'm pretty negative about casting everything to masked arrays. If anything I'd go the other way and strip the mask and enter NaN.

I understand the concerns with masked arrays but I do think we should include this change into the library mainly because of consistency and user expectations. Sure, we use the `np.ma` functions only if the input is masked but the behaviour shouldn't be that we lose the mask entirely, imo (as it currently happens).

Reply:
@ksunden , just pinging you here for background since you're working on the new data and units model. The `MaskedArray` handling throughout the code is something you might want to look at as well.